### ISSUE TITLE
Candidate window is not shown in Windows

### ISSUE THREAD

**Issue by yukawa** [google/mozc#788]
## Description
In Windows, UI components that are implemented by `mozc_renderer`, e.g. candidate window, will not be shown.

## Steps to reproduce
1. Build Mozc and install it into Windows 11 64-bit environment
2. Open notepad
3. Select Mozc
4. Enable Mozc
5. Try to type anything

## Expected behavior
At the step 4, you can see a mode indicator around the cursor.
At the step 5, you can see a suggest window around the cursor.

## Actual behavior
At the step 4, you don't see the mode indicator around the cursor.
At the step 5, you don't see the suggest window around the cursor. Text conversion still works though.

## Version or commit-id
c2ccb4f6a65ddda244cfa103d8cdfe6aaeaf5eda

## Environment
 - OS: Windows 11
 - Related Applications: Any app

## Additional context
This is a regression introduced in e7e8fb29b9bd671cc22cbe8efefdae95162dc722.

What's actually happening is that `mozc_renderer` is immediately crashing here because `render_font` is `nullptr`.
https://github.com/google/mozc/blob/c2ccb4f6a65ddda244cfa103d8cdfe6aaeaf5eda/src/renderer/win32/text_renderer.cc#L344

The reason why `render_font` became `nullpter` is a behavior change in `CreateFormat` here.
https://github.com/google/mozc/commit/e7e8fb29b9bd671cc22cbe8efefdae95162dc722#diff-72ab72c6eb41fd09771591806ce7d1515cc5ea094ee533cb1da5fd03d5407610L487-R492

In short, a `wchar_t` buffer passed to [`IDWriteLocalizedStrings::GetString()`](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritelocalizedstrings-getstring) needs to be 1 element bigger than the length returned from [`IDWriteLocalizedStrings::GetStringLength()`](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nn-dwrite-idwritelocalizedstrings) so that `\0` can also be stored. e7e8fb29b9bd671cc22cbe8efefdae95162dc722 accidentally removed such a special handling, which ended up causing an access violation in `mozc_renderer`.
